Utils/azuretools-core/src/com/microsoft/azuretools/adauth/AuthContext.java

```diff
@@ -97,13 +86,17 @@ public String getClientId() {
      * Get token from last authentication result
      *
      * @param lastResult last authentication result
+     * @param webUi web ui
+     * @param redirectUrl redirect url
      * @return authentication result with updated tokens
      * @throws AuthException exception during getting token
      */
-    public AuthResult acquireToken(@NotNull AuthResult lastResult) throws AuthException {
+    public AuthResult acquireToken(@NotNull final AuthResult lastResult, final IWebUi webUi,
```

Comment:
If the `webUi` and `redirectUrl` are only used for Device login, can they be put into DCAuthManager? such as constructor of DCAuthManager

Reply:
They are only used in interactive login.